### ISSUE TITLE
Upload not works on android wechat

### ISSUE THREAD

**Issue by dfang** [moxiecode/plupload#1397]
on iOS Uploading works very well, but on android wechat within some wifi network,` FilesAdded`, `BeforeUpload`, `UploadFile` and ` UploadProgress`events triggerd but `FileUploaded` didn't. there's even no request on server logs, but switch to connect with my iOS hotspot, it works well.

it's really really weird !  here is the config:
```
webUploader = new plupload.Uploader({
      runtimes        : 'html5, flash, html4',
      browse_button   : pickerId,
      max_file_size   : '10mb',
      url             : uploadUrl,
      flash_swf_url   : '/uploader.swf',
      filters         : [ { title : "图片文件", extensions : "jpg,jpeg,gif,png,bmp" } ],
      file_data_name  : 'file',
      multipart       : true,
      multi_selection : false,
      dragdrop        : false,
      max_retries     : 3,
      multipart_params: { authenticity_token: $('meta[name="csrf-token"]').attr('content') }
    })
```

**Comment by suhaotian** [comment]
Same here

**Comment by colorwin** [comment]
+1

**Comment by suhaotian** [comment]
Hi, guys, Not working in wechat ？ please see here : https://www.zhihu.com/question/54623070


@colorwin @dfang

**Comment by colorwin** [comment]
@suhaotian 
I use version 2.3, but I can't find this line:
```
"image/jpeg,jpg jpeg jpe," +
```
I find 3.x has this line, but it seem not compatible to my project.

**Comment by suhaotian** [comment]
@colorwin  Here:
![image](https://user-images.githubusercontent.com/6512574/27256841-25f696ca-53f5-11e7-9b19-043ba6d8c1b6.png)

**Comment by colorwin** [comment]
@suhaotian  Thanks, I find it.

**Comment by jayarjo** [comment]
Did adding `image/jpg` mime-type solve the problem?

**Comment by jayarjo** [comment]
@colorwin sorry not familiar with wechat is it messaging app? does it use Plupload to upload something?

**Comment by colorwin** [comment]
@jayarjo 
Yeah, [wechat](https://www.wikiwand.com/en/WeChat) is a messaging app in China like whatsapp, it can browse web in inner app. It use itself X5 blink which seem based on webkit. 

Some android phones user  report some upload problems in wechat's inner browser(I use Plupload to do upload job):
1. Click the file button but no response.(major)
2. No response after file choose.
3. Can't use system album just can use picture.

But I also can't appear these issue. After adding `image/jpg mime-type` , the report seem become less, but still have. At the last, my boss transfer the upload file job to APP, so I am sorry to know more.

Last but not least, my boss guess that some ardroid phone are too old to tune up andriod webview. I am in China, but our user are major in foreign country.  Network problem?

**Comment by jayarjo** [comment]
`image/jpg` is kind of invalid mime-type, that's why I was wondering. Can you specify an exact platform that has these problems all the time? So that we could replicate it.

**Comment by suhaotian** [comment]
@jayarjo It's wechat's webview bug, not android ~

**Comment by jayarjo** [comment]
@suhaotian so wechat's webview bug for some reason requires invalid jpeg mime type (`image/jpg`)?

Also do you know if this happens with other file types, or only - jpegs?

**Comment by suhaotian** [comment]
Yes,  wechat's webview bug for some reason requires invalid jpeg mime type (image/jpg).

 I can't answer  second question, Maybe tonight go home test it with `.png` ? hh

**Comment by Dragon-Rider** [comment]
@suhaotian Thanks for sharing :).
I met a problem on android wechat. The wechat webview pop up a toast of "No apps can perform this action."  I left this hint for the rest people who have the same problems: P
